sdk/python/tests/unit/permissions/auth/test_token_parser.py

```diff
@@ -63,6 +63,58 @@ def test_oidc_token_validation_failure(mock_oauth2, oidc_config):
         )
 
 
+@mock.patch.dict(os.environ, {"INTRA_COMMUNICATION_BASE64": "test1234"})
+@pytest.mark.parametrize("intra_communication_val", ("test1234", "my-name"))
+def test_oidc_inter_server_comm(intra_communication_val, oidc_config, monkeypatch):
+    async def mock_oath2(self, request):
+        return "OK"
+
+    monkeypatch.setattr(
+        "feast.permissions.auth.oidc_token_parser.OAuth2AuthorizationCodeBearer.__call__",
+        mock_oath2,
+    )
+    signing_key = MagicMock()
+    signing_key.key = "a-key"
+    monkeypatch.setattr(
+        "feast.permissions.auth.oidc_token_parser.PyJWKClient.get_signing_key_from_jwt",
+        lambda self, access_token: signing_key,
+    )
+
+    user_data = {
+        "preferred_username": f"{intra_communication_val}",
+    }
+
+    if intra_communication_val != "test1234":
```

Comment:
instead of using this check, you can add another pytest parameter with a boolean defining the expected behavior, like `is_intra_server`

Reply:
> instead of using this check, you can add another pytest parameter with a boolean defining the expected behavior, like `is_intra_server`

I need the actual value in the environment variable to compare it its not enough to know that the the environment variable is set but it must be the same in the client and server

Reply:
so you don't need to know the env variable, you set the value in the parameter